### PR TITLE
Include runtime deps in test archive

### PR DESCRIPTION
**Pull request by oxve**
A collection of small updates that fixes failing unit tests:
* Include runtime_deps files in test archive
* Disable annotations for junit parser (increases parsing speed)
* Continue running tests after suite fails, print failing suites at the end

b/372303096
b/371590965

### CODE REVIEW

File: cobalt/docker/linux/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,30 @@
 # Use the official Ubuntu image as the base
 FROM debian:12
 
+# TODO(b/372303096): Move to dedicated unittest image.
+RUN echo deb http://deb.debian.org/debian-debug \
+    $(. /etc/os-release && echo "${VERSION_CODENAME}")-debug main \
+    >> /etc/apt/sources.list
+
+# TODO(b/372303096): Move to dedicated unittest image.
```
Comment (isarkis) [inline cobalt/docker/linux/Dockerfile:9]: nit: can we add this TODO to an actual code vs having to maintain 2 lists?

**Review by isarkis — APPROVED** [review]

File: cobalt/docker/linux/Dockerfile (hunk 1 above)
Comment (oxve) [reply cobalt/docker/linux/Dockerfile:9]: Best I can do is to add a separate apt command for these. Do you prefer that?

Can't put comments in those escaped lines as bash will concat them and everything after the first comment is now a comment.

Comment (isarkis) [reply cobalt/docker/linux/Dockerfile:9]: ah never mind then, I am was just afraid two lists will go out of sync.

File: .github/actions/process_test_results/action.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ runs:
       uses: mikepenz/action-junit-report@992d97d6eb2e5f3de985fbf9df6a04386874114d
       with:
         report_paths: results/*.xml
-        annotate_only: true
+        skip_annotations: true
```
Comment (imindich) [inline .github/actions/process_test_results/action.yaml:24]: Did this end up significantly reducing total time for the publisher? Do we still want to replace with a different one?

Comment (oxve) [reply .github/actions/process_test_results/action.yaml:24]: It did speed it up ~10x but that is still longer than the time it took to actually run the tests. I think we should try to find a replacement still.

**Review by imindich — APPROVED** [review]

**Pull request merged**